Gems/ROS2/Code/Source/Camera/ROS2CameraSensorComponent.h

```diff
@@ -79,19 +72,20 @@ namespace ROS2
             return { publisher, AZStd::make_shared<CameraType>(description) };
         }
 
+        //! Helper that adds an image source.
+        //! @param imagePostProceessingFn a function to be called for post processing.
+        //! @param depth if true, depth image source is added, else it is a color image source.
+        void AddImageSource(const Camera::ImagePostProcessingFn& imagePostProceessingFn, bool depth = false);
```

Comment:
comment: I have a pet peeve against `bool` parameters 😅 Feel free to ignore me but to make the callsite clearer it might be worth making this an `enum` instead we can pass in (or make two functions called `AddImageSourceColor` and `AddImageSourceDepth` respectively which delegate to a shared internal/private function that can use a `bool`.

Reply:
Agree with Tom, I propose using a template argument.